cpp_exercises/day2/date/date.cpp

```diff
@@ -14,3 +14,8 @@ bool operator==(const Date& dateA, const Date& dateB) {
 bool Date::isLeapYear(int year) {
     return year % 100 == 0? (year % 400 == 0):(year % 4 == 0);
 }
+
+int Date::getMonthDays(int month, int year) {
+    return (Date::isLeapYear(year) && month == Date::FebruaryZeroBase) ? (Date::m_monthsDays[month] + 1) : (Date::m_monthsDays[month]) ;
```

Comment:
Qui stiamo facendo ogni volta il check sull'anno bisestile, quando invece meriterebbe farlo solo dopo aver appurato che il mese è quello di febbraio: possiamo invertire le condizioni in AND dell'if e risparmiarci una chiamata a funzione nel 92% delle occasioni.
In questi casi preferisco usare l'[happy-path](https://en.wikipedia.org/wiki/Happy_path) per il ritorno, mettendo in evidenza i vari casi speciali durante il flusso di codice, e inserendo il valore che nella maggior parte dei casi è quello giusto alla fine. 
`
if (month == Date::FebruaryZeroBase)
    return Date::m_monthsDays[month] + Date::isLeapYear(year);
return Date::m_monthsDays[month];
`